packages/sku/src/services/vite/plugins/skuVitePreloadPlugin.ts

```diff
@@ -0,0 +1,150 @@
+      return Boolean(config.build?.ssr);
+    },
+
+    async transform(code, id) {
+      if (!include.test(id)) {
+        return null;
+      }
+      let ast: ReturnType<typeof parse>;
+
+      // TODO: Do some better checking. Not every function thats called `loadable` is the correct one.
+      // TODO: Make sure that the import is correct as well to failsafe.
+      // Find dynamic imports
+      if (code.includes(' import(')) {
+        ast = parse(code, {
+          sourceType: 'module',
+          plugins: ['jsx', 'typescript'],
+        });
+
+        let injected = false;
+
+        traverse(ast, {
+          CallExpression(callPath) {
+            if (
+              callPath
+                .get('callee')
+                .isIdentifier({ name: loadablefunctionName })
+            ) {
+              // TODO: This can be cleaned up. perhaps not doing filesystem reads to get the file ext.
+              const importArg = callPath.get('arguments')[0];
+              importArg.traverse({
+                Import(importPath) {
+                  // @ts-expect-error
+                  if (!importPath.parent.arguments) {
+                    return;
+                  }
+                  // @ts-expect-error
+                  const importArgument = importPath.parent.arguments[0];
+
+                  if (importArgument) {
+                    // Dynamic import of a dynamic module is not supported
+                    if (importArgument.type === 'StringLiteral') {
+                      const absolutePath = resolve(
+                        dirname(id),
+                        importArgument.value,
+                      );
+                      const files = readdirSync(dirname(absolutePath));
+                      const name = pathParse(absolutePath).base;
+
+                      const found = files.find(
+                        (x) =>
+                          x.replace(extname(x), '') ===
+                          name.replace(extname(name), ''),
+                      );
+                      const relativePath = getRelativePath(
+                        `${dirname(absolutePath)}/${found}`,
+                      );
+
+                      callPath.node.arguments.push(
+                        t.stringLiteral(relativePath),
+                      );
+                      injected = true;
+                    }
+                  }
+                },
+              });
+            }
+          },
+        });
+
+        if (injected) {
+          if (debug) {
+            this.info('Injected moduleId in React component');
+          }
+          count++;
+          const output = generate(ast, {}, code);
+          injectedModules.add(id);
+          return {
+            code: output.code,
+            map: output.map,
+          };
+        }
+      }
+
+      return null;
+    },
+
+    buildEnd() {
+      if (debug) {
+        const s = lazyImportedModules.difference(injectedModules);
```

Comment:
Worth noting that `Set.difference` support was added [in Node 22](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/difference#browser_compatibility), which is above sku's current minimum supported node version. That being said, we could also just require node v22 when using sku's vite mode. It's experimental, so we can do whatever we want 😎.